### PR TITLE
ARROW-12333: [JS] Remove jest-environment-node-debug and do not emit from typescript by default

### PR DESCRIPTION
**Pull request by domoritz**

### CODE REVIEW

File: js/tsconfig.json [hunk 1]
```diff
@@ -6,6 +6,7 @@
   },
   "compilerOptions": {
     "target": "ESNEXT",
-    "module": "commonjs"
+    "module": "commonjs",
+    "noEmit": true
```
Comment (domoritz) [inline js/tsconfig.json:10]: Now we can run yarn tsc to check types without accidentally writing a bunch of files.

File: js/package.json [hunk 2]
```diff
@@ -86,7 +86,6 @@
     "gulp-typescript": "5.0.1",
     "ix": "2.5.3",
     "jest": "26.3.0",
-    "jest-environment-node-debug": "2.0.0",
```
Comment (domoritz) [inline js/package.json:89]: I didn't see it being used anywhere.

Comment (trxcllnt) [reply js/package.json:89]: yeah I think this used to be necessary for debugging, but has since been fixed in Jest

**Review by trxcllnt — APPROVED** [review]

File: js/package.json (hunk 2 above)
Comment (domoritz) [reply js/package.json:89]: Excellent.

**Pull request closed without merging**